appc/appconnector.go

```diff
@@ -125,10 +176,26 @@ func (e *AppConnector) updateDomains(domains []string) {
 		for _, wc := range e.wildcards {
 			if dnsname.HasSuffix(d, wc) {
 				e.domains[d] = addrs
+				delete(oldDomains, d)
 				break
 			}
 		}
 	}
+
+	// Everything left in oldDomains is a domain we're no longer tracking
+	// and if we are storing route info we can unadvertise the routes
+	if e.ShouldStoreRoutes() {
+		toRemove := []netip.Prefix{}
+		for _, addrs := range oldDomains {
+			for _, a := range addrs {
+				toRemove = append(toRemove, netip.PrefixFrom(a, a.BitLen()))
```

Comment:
Optional: if `e.Domains` / `oldDomains` had a value type of `[]netip.Prefix` then the inner loop could be replaced with `toRemove = append(toRemove, addrs...)` which may also have fewer allocations.

`toRemove = slices.Grow(toRemove, len(addrs))` before the inner loop wouldn't hurt. Up to you if you want to do that or even count the total number in a first pass.